server/fleet/app.go

```diff
@@ -166,17 +167,66 @@ type AppConfig struct {
 
 	// VulnerabilitySettings defines how fleet will behave while scanning for vulnerabilities in the host software
 	VulnerabilitySettings VulnerabilitySettings `json:"vulnerability_settings"`
+
+	WebhookSettings WebhookSettings `json:"webhook_settings"`
+}
+
+type Duration struct {
+	time.Duration
+}
+
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(d.String())
+}
+
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		d.Duration = time.Duration(value)
+		return nil
+	case string:
+		var err error
+		d.Duration, err = time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		return nil
+	default:
+		return errors.New("invalid duration")
+	}
+}
+
+type WebhookSettings struct {
+	HostStatusWebhook HostStatusWebhookSettings `json:"host_status_webhook"`
+	Periodicity       Duration                  `json:"periodicity"`
+}
+
+type HostStatusWebhookSettings struct {
+	Enable         bool    `json:"enable_host_status_webhook"`
+	DestinationURL string  `json:"destination_url"`
+	HostPercentage float64 `json:"host_percentage"`
+	DaysCount      int     `json:"days_count"`
 }
 
 func (ac *AppConfig) ApplyDefaultsForNewInstalls() {
 	ac.ServerSettings.EnableAnalytics = true
-	ac.HostSettings.EnableHostUsers = true
 	ac.SMTPSettings.SMTPPort = 587
 	ac.SMTPSettings.SMTPEnableStartTLS = true
 	ac.SMTPSettings.SMTPAuthenticationType = AuthTypeNameUserNamePassword
 	ac.SMTPSettings.SMTPAuthenticationMethod = AuthMethodNamePlain
 	ac.SMTPSettings.SMTPVerifySSLCerts = true
 	ac.SMTPSettings.SMTPEnableTLS = true
+
+	ac.ApplyDefaults()
+}
+
+func (ac *AppConfig) ApplyDefaults() {
```

Comment:
I realized we don't have "defaults to set if the json doesn't have the key", which is what we might want as we add new configs and people upgrade. So I added this and called this before unmarshaling when we call `ds.AppConfig()`